### DISCUSSION TITLE
puppet-control-repo classes missing specs

### DISCUSSION THREAD

**Discussion by cdenneen** [puppetlabs-toy-chest/pct#294] [Feedback]
So the `pct new puppetlabs/puppet-class` creates matching specs for each class but the base/example profiles and roles that come with a `puppetlabs/puppet-control-repo` is missing those files and even more the opinionated way to test those profile/roles. 

Providing those examples help people contruct good way to test.

**Comment by michaeltlombardi** [comment]
This is a good shout @cdenneen - unfortunately, no one on the team maintains any control repos, but we'd be happy to convert this to an issue for tracking and further discussion/implementation. If you have anything in particular you'd like to see for those test files, we'd be happy to pair with you (and if not, we'll do the research and put this on our to-do).

**Reply by cdenneen** [reply]
I think similar to the "opinionated" way stubbing out tests for modules it would be useful to stub out testing the example profile/roles in a control repo. Give example of an "opinionated" way to do so.  I've seen people write them inside profile/role treating them as modules and I've seen early examples where the tests are at the `control-repo/spec` level `(control-repo/spec/classes/profile,role)`. 
The fact that pdk/pct is helping provide users the way to structure control-repos is great but it would be useful to see how is best (or at least an opinionated recommentation) to test those.
This would allow for stubbing of rspec testing of those and even litmus examples of how to test as well.
